examples/solverdummies/c/solverdummy.c

```diff
@@ -37,37 +44,73 @@ int main(int argc, char **argv)
 
   meshID = precicec_getMeshID(meshName);
 
-  dimensions = precicec_getDimensions();
-  vertex     = malloc(dimensions * sizeof(double));
+  if (strcmp(participantName,"SolverOne") == 0){
+    writeDataID = precicec_getDataID("dataOne",meshID);
+    readDataID = precicec_getDataID("dataTwo",meshID);
+  }
+  if (strcmp(participantName,"SolverTwo") == 0){
+    writeDataID = precicec_getDataID("dataTwo",meshID);
+    readDataID = precicec_getDataID("dataOne",meshID);
+  }
```

Comment:
It is a bit strange to use `participantName` as a "key" for all the parameters (`writeDataID`, `readDataID`) except `meshName`. I would suggest something similar to the pesudocode below: 

```
  int meshID;
  const char *meshName;
  if (strcmp(participantName,"SolverOne") == 0){
    meshName = "meshOne";
    meshID = precicec_getMeshID(meshName);
    writeDataID = precicec_getDataID("dataOne",meshID);
    readDataID = precicec_getDataID("dataTwo",meshID);
  }
  if (strcmp(participantName,"SolverTwo") == 0){
    meshName = "meshTwo";
    meshID = precicec_getMeshID(meshName);
    writeDataID = precicec_getDataID("dataTwo",meshID);
    readDataID = precicec_getDataID("dataOne",meshID);
  }
  assert(strcmp(argv[3], meshName));
```

I would suggest to not change the signature, since this means we will have to change all our tests, as well. The assertion ensures that everything is still consistent. Let's clean this up later.

If we want to merge this PR now, I would also be totally fine with turning all this into an issue and leave it for now as it is.

Reply:
We decided not to do the suggested change above, but we will directly solve the problem thoroughly in the future (https://github.com/precice/precice/issues/829).